Cabal/src/Distribution/Simple/Errors.hs

```diff
@@ -210,8 +208,6 @@ exceptionCode e = case e of
   NoSupportBuildingTestSuite{} -> 4106
   NoSupportBuildingBenchMark{} -> 5320
   BuildingNotSupportedWithCompiler{} -> 7077
-  ProvideHaskellSuiteTool{} -> 7509
-  CannotDetermineCompilerVersion{} -> 4519
```

Comment:
Do you not need to keep these error codes? Since what in future if someone adds another 4519 error which clashes with older versions of Cabal?

I admit, this is probably a theoretical question.

Reply:
This is a good point; we should have a way to mark obsolete error numbers as such.

Reply:
I checked the [webpage that resolves these errors](https://errors.haskell.org/) to descriptions, and there were none for these two errors, so I assumed I can remove them without comment.
(The only Cabal error defined there yet is 7070.)

Reply:
Do you think a `-- Retired.` comment would do (leaving the datatype intact and the constructor there) or would you prefer something like

```
data WrappedExpection =
     RetiredException Int
  |  LiveException CabalException
```

?

Reply:
I think a comment would do for now, GHC has a much more elaborate scheme in place to ensure that codes are not reused. (See https://gitlab.haskell.org/ghc/ghc/-/merge_requests/8849)

Reply:
@mpickering @geekosaur @ffaf1 : Is this patch addressing the concerns?

Reply:
Fine by me!

Reply:
Possibly factor that question out into a separate issue for the future, but this one is fine.

Reply:
Although I will agree with "documented on the errors webpage" not being the right concern, given that nothing else is either. (I think we have an issue for that, and it might be GSoC-worthy to get that changed.)

Reply:
Yes, I don't object in this instance, but it should be considered in future.